File: codeforces/B_Robots.py
# https://codeforces.com/problemset/problem/1680/B

# Time complexity: O(N*M)
# Space complexity: O(1) Constant time complexity
def solve(n, m, matrix):
    min_point = None
    min_row = n
    min_column = m
    for i, row in enumerate(matrix):
        for j, cell in enumerate(row):
            if cell == 'R':
                if not min_point:
                    min_point = (i, j)
                min_row = min(min_row, i)
                min_column = min(min_column, j)
                if min_point != (min_row, min_column):
                    return 'NO'
    return 'YES'


t = int(input())
for _ in range(t):
    n, m = [int(e) for e in input().split()]
    matrix = []
    for _ in range(n):
        matrix.append(input())
    print(solve(n, m, matrix))

File: codeforces/B_Robots.java
// https://codeforces.com/problemset/problem/1680/B
import java.util.Scanner;

public class B_Robots {

    // Time complexity: O(N*M)
    // Space complexity: O(1) Constant time complexity
    public static String solve(int n, int m, String[] matrix){
        int[] min_point = {-1, -1};
        int min_row = n;
        int min_column = m;
        for(int i = 0; i<n; i++){
            for (int j = 0; j < m; j++) {
                char cell = matrix[i].charAt(j);
                if(cell == 'R'){
                    if(min_point[0] == -1 && min_point[1] == -1){
                        min_point[0] = i;
                        min_point[1] = j;
                    }
                    if(i < min_row)
                        min_row = i;
                    if(j < min_column)
                        min_column = j;
                    if(!(min_point[0] == min_row && min_point[1] == min_column)){
                        return "NO";
                    }
                }
            }
        }
        return "YES";
    }

    public static void main(String[] args) {
        Scanner lector = new Scanner(System.in);
        int t = lector.nextInt();
        for(int i = 0; i< t; i++){
            int n = lector.nextInt();
            int m = lector.nextInt();
            String[] matrix = new String[n];
            for(int j = 0; j<n; j++){
                matrix[j] = lector.next();
            }
            System.out.println(solve(n, m, matrix));
        }
    }
}
